### DISCUSSION TITLE
Add possibility to get notified when tasks changes state during its lifecycle

### DISCUSSION THREAD

**Discussion by hcf** [kagkarlsson/db-scheduler#448] [Ideas]
When using `oneTimeTasks` in a manner similar to a job queue, it would be useful to have some insight into the lifecycle of the tasks in order to act on this in our application, for instance show users that a long running process is happening in the background.

I have tried to find an existing way of doing this in `db-scheduler`, but the closest I can find is the `StatsRegistry` interface. It feels hacky though, because there is a clear difference between gathering stats and listening to change events, and if we both want stats and change events there is no possibility of this.

**Comment by kagkarlsson** [comment]
Hans-Christian! Good to see you! 😄 

For the specific case of long-running processes (current, not historic), you may use `scheduler.getCurrentlyExecuting()` to get some insight as to what is running on this particular scheduler-instance and when it started running. Pretty sure I have previously used that to create Micrometer gauges for long-running tasks.

As to `StatsRegistry`, you might also use it as you suggest. I have seen similar needs elsewhere, and I think it makes sense to evolve it into a Listener-type interface instead, and also for some events, send along more data (the execution). Cannot say when that will happen though! :)
